src/lib/array.jl

```diff
@@ -957,3 +959,15 @@ end
   back(Δ::AbstractArray) = (nothing, getindex.(_back.(Δ), 1))
   return Fill(y, size(r)), back
 end
+
+# Sparse Arrays
+
+@adjoint function SparseMatrixCSC{T,N}(arr) where {T,N}
+  SparseMatrixCSC{T,N}(arr), Δ -> (collect(Δ),)
+end
+@adjoint diagm(x) = diagm(x), Δ -> (diag(Δ), )
+
+@adjoint function Broadcast.broadcasted(Float32, a::SparseMatrixCSC{T,N}) where {T,N}
+  Float32.(a), Δ -> (nothing, T.(Δ), )
+end
+@adjoint issymmetric(x) = issymmetric(x), Δ -> (Δ,)
```

Comment:
Is this differentiable, or should it be `@nograd`ed?

Reply:
I think it needs the adjoint here, else you'll get a bunch of `nothing`s

Reply:
What's wrong with getting `nothing`s in this context?

Reply:
It can lead to dropped gradients for `x` then I think

Reply:
Looking at this another way, `Δ` here is presumably a scalar of some kind, so isn't an appropriate gradient for `x`.

Reply:
Similarly, `issymmetric(randn())` will always return `true`, so the derivative should definitely be some flavour of `0`.

Reply:
Done